prov/tcp/src/tcpx_ep.c

```diff
@@ -209,18 +254,31 @@ static int tcpx_pep_sock_create(struct tcpx_pep *pep)
 		return -FI_EIO;
 	}
 
-	ret = tcpx_setup_socket(pep->sock);
-	if (ret) {
-		goto err;
-	}
+	if (ofi_addr_get_port(pep->info->src_addr) != 0 || port_range.high == 0) {
+		ret = tcpx_setup_socket(pep->sock);
+		if (ret) {
+			goto err;
+		}
+		ret = bind(pep->sock, pep->info->src_addr,
+		      (socklen_t) pep->info->src_addrlen);
+
+		if (ret) {
+			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+				"failed to bind listener: %s\n",
+				strerror(ofi_sockerr()));
+			goto err;
+		}
+	} else {
+		ret = tcpx_setup_socket_nodelay(pep->sock);
+		if (ret) {
+			goto err;
+		}
 
-	ret = bind(pep->sock, pep->info->src_addr,
-		   (socklen_t) pep->info->src_addrlen);
-	if (ret) {
-		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
-			"failed to bind listener: %s\n",
-			strerror(ofi_sockerr()));
-		goto err;
+		ret = tcpx_bind_to_port_range(pep->sock, pep->info->src_addr,
+					      pep->info->src_addrlen);
+		if (ret){
+			goto err;
+		}
```

Comment:
Drop this if statement and use the one above moved down, so we get an error message in either case.

Reply:
Done.